pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -26,13 +30,28 @@ func (ca *cloudCIDRAllocator) performMultiNetworkCIDRAllocation(node *v1.Node, i
 	if err != nil {
 		return nil, fmt.Errorf("node=%s error fetching networks: %v", node.Name, err)
 	}
+
+	// get networks from Node's network-status annotation
+	nodeUpNet, err := getUpNetworks(node)
+
+	// statusNetworks is map of networks that are Ready and status==UP
+	statusNetworks := make(map[string]struct{})
+	// networks is list of networks that are Ready
 	networks := make([]*networkv1.Network, 0)
-	// ignore networks that are under deletion.
+	// filter networks based on Ready condition
 	for _, network := range k8sNetworksList {
-		if network.DeletionTimestamp.IsZero() {
```

Comment:
you may want to keep skipping the networks that are being deleted
```
if !network.DeletionTimestamp.IsZero() {
   continue
```

Reply:
We are shifting to using the network-status annotation as the source of truth for whether a network is up or down as opposed to watching the deletion timestamp. In this case, we could technically use the deletion timestamp to bypass the source of truth at the current moment, but the dependencies for a network being up or down status may change in the future and this keeps things simple as all components rely on status.

Reply:
Evan, is it the case that if the network is being deleted the NetworkConditionStatusReady will be false? Otherwise we could do the optimization Antonio is suggesting.

Reply:
NetworkConditionStatusReady wont be false if the network is being deleted. We could do the optimization now. If something changes in the future where we need to keep the annotation until network status changes we could change the behavior here.

Reply:
added